stdlib/src/utils/span.mojo

```diff
@@ -224,3 +224,20 @@ struct Span[
         """
 
         return self._data
+
+    @always_inline
+    fn copy_from[
+        lifetime: MutableLifetime, //
+    ](ref [_]self: Span[T, lifetime], ref [_]other: Span[T, _]):
+        """
+        Performs an element wise copy from all elements of `other` into all elements of `self`.
+
+        Parameters:
+            lifetime: The inferred mutable lifetime of the data within the Span.
+
+        Args:
+            other: The Span to copy all elements from.
+        """
+        debug_assert(len(self) == len(other), "Spans must be of equal length")
+        for i in range(len(self)):
```

Comment:
Should I be doing a `memcpy` instead of iterating in a loop?

Reply:
You're doing the right thing🙂 A `memcpy` would not be semantically valid for many types.

Conceptually and literally you're moving these types in memory, so calling `__moveinit__` here is the right thing to do.

For example, consider a type whose `__moveinit__` constructor contains custom logic when moving the type in memory, e.g. to fixup a self-referential pointer. Failing to call `T.__moveinit__` when moving it in memory would break the invariants of that type.

Though, all that being said, using `memcpy()` as an _optimization_ if and only if `T: <TriviallyCopyable>` would be a great thing to do, but we can not currently express that constraint properly in Mojo yet. (Mojo does not have an equivalent to the Rust [`Copy`](https://doc.rust-lang.org/stable/std/marker/trait.Copy.html) marker trait. `AnyTrivialRegType` is _similar_ but not quite usable at the moment for specialization in the same way.)